### PR TITLE
Include errorinfo in email_archive.

### PR DESCRIPTION
**Pull request by neekfenwick**
Fixes #6449 

*PLEASE REVIEW* the database upgrade script, I've had a half hearted stab at it.  I still don't know how this install/upgrade process works and I'm surprised to see the latest file has version `210` in it.  My changes here probably need rewriting and including in the install script too.

I've tested this change on my local zencart, running the `ALTER TABLE` manually.  It inserts NULL errorinfo if no errorinfo was provided (empty string), otherwise the errorinfo stirng.

Also rewrote the email_archive insert to use `$db->perform` just because I feel it looks neater.  Happy to stand corrected.

### CODE REVIEW

**Comment by drbyte** [comment]
> I'm surprised to see the latest file has version 210 in it. 

The `master` branch is the work-in-progress branch preparing for the next X.Y release.  
Anyone building something to go into 2.1.0 will need a place to put those upgrade additions, so the 210 file was added for that purpose.

(`master` is latest unreleased dev code;  `2.0` branch is bug-fixes-against v2.0.X, so will form 2.0.0, 2.0.1, 2.0.2, etc until 2.1.0 is released and a `2.1` branch is created for that series of bugfixes)

File: zc_install/sql/updates/mysql_upgrade_zencart_210.sql [hunk 1]
```diff
@@ -34,7 +34,9 @@
 TRUNCATE TABLE whos_online;
 TRUNCATE TABLE db_cache;
 
-
+#PROGRESS_FEEDBACK:!TEXT=Adding error logging to email archive ...
+# Add column to store any errorinfo returned from phpmailer.
+ALTER TABLE ADD ALTER TABLE email_archive ADD COLUMN (errorinfo TEXT NULL);
```
Comment (drbyte) [inline zc_install/sql/updates/mysql_upgrade_zencart_210.sql:39]:
```suggestion
ALTER TABLE email_archive ADD COLUMN errorinfo TEXT DEFAULT NULL;
```

**Comment by drbyte** [comment]
Should also add the extra field in the main install SQL. After the `module` field, and before the `PRIMARY KEY` stuff.
https://github.com/zencart/zencart/blob/7cc02d56f1ebb816d69608900e60cb17814569c7/zc_install/sql/install/mysql_zencart.sql#L737-L760

**Comment by drbyte** [comment]
> It inserts NULL errorinfo if no errorinfo was provided (empty string), otherwise the errorinfo stirng.

Excellent.

> Also rewrote the email_archive insert to use $db->perform just because I feel it looks neater. 

Indeed.

File: zc_install/sql/updates/mysql_upgrade_zencart_210.sql (hunk 1 above)
Comment (neekfenwick) [reply zc_install/sql/updates/mysql_upgrade_zencart_210.sql:39]: Wow that was messy :) My lazy copy paste. Thanks.

**Comment by neekfenwick** [comment]
> Should also add the extra field in the main install SQL. After the `module` field, and before the `PRIMARY KEY` stuff.

Done, thanks.

**Pull request merged**